docs/reference/filter_field_definition.rst

```diff
@@ -66,8 +66,27 @@ Example
         }
     }
 
-doctrine_orm_model_autocomplete
--------------------------------
+ArrayFilter
+-----------
+
+This filter is made to filtering on values saved in databases as a serialized arrays of string with the
+``@ORM\Column(type="array")`` annotation. It is recommended to use another table and ``OneToMany`` relations
+if you want to make complex ``SQL`` queries or if your table is too big and you get performance issue but
+this filter can provide some basic queries::
+
+    protected function configureDatagridFilters(DatagridMapper $datagridMapper): void
+    {
+        $datagridMapper
+            ->add('keywords', ArrayFilter::class);
+    }
+
+.. note::
+
+    The filter can give bad results with associative arrays since it is not easy to distinguish between keys
+    and values for a serialized associative array.
```

Comment:
Add also a warning about values that are `s` or `b` or whatever can confuse this? BTW, maybe this would be less risky with json…

Reply:
I don't understand why `s` or `b` can confuse this @greg0ire. 
I serialize the value you're looking for. So if you're looking for `s`, it'll search
`"s:1:"s";"` this can't be confused with a `s` in a bigger word or the `s` used by the serialize array to say `string`.

The only possible confusion is when you store string which are already contain serialize value like storing the array `['s:4:"toto";', 'toto']`. But this seems to be a really weird case. Should I mention it ?

It does not seems to me easy to migrate a `@ORM\Column(type="array")` to a `@ORM\Column(type="json")`, isn't it ? So I would prefer to offer a filter for `array` and a filter for `json` (or the same filter with a `type` option).

Reply:
Oh right, `s` and `b` look safe to search for indeed, my bad.